src/test/java/org/opensearch/security/configuration/ConfigurationRepositoryTest.java

```diff
@@ -0,0 +1,146 @@
+    }
+
+    @Test
+    public void initOnNodeStart_withSecurityIndexNotCreatedShouldNotSetInstallDefaultConfig() {
+        Settings settings = Settings.builder().put(ConfigConstants.SECURITY_BACKGROUND_INIT_IF_SECURITYINDEX_NOT_EXIST, false).build();
+
+        ConfigurationRepository configRepository = createConfigurationRepository(settings);
+
+        configRepository.initOnNodeStart();
+
+        assertThat(configRepository.getInstallDefaultConfig().get(), is(false));
+    }
+
+    @Test
+    public void getConfiguration_withInvalidConfigurationShouldReturnNewEmptyConfigurationObject() throws IOException {
+        ConfigurationRepository configRepository = createConfigurationRepository(Settings.EMPTY);
+
+        SecurityDynamicConfiguration<?> config = configRepository.getConfiguration(CType.CONFIG);
+        SecurityDynamicConfiguration<?> emptyConfig = SecurityDynamicConfiguration.empty();
+
+        assertThat(config, instanceOf(SecurityDynamicConfiguration.class));
+        assertThat(config.getCEntries().size(), is(equalTo(0)));
+        assertThat(config.getVersion(), is(equalTo(emptyConfig.getVersion())));
+        assertThat(config.getCType(), is(equalTo(emptyConfig.getCType())));
+        assertThat(config.getSeqNo(), is(equalTo(emptyConfig.getSeqNo())));
+        assertThat(config, is(not(equalTo(emptyConfig))));
+    }
+
+    @Test
+    public void getConfiguration_withValidConfigurationShouldReturnDeepClone() throws Exception {
+
+        Settings settings = Settings.builder().put(ConfigConstants.SECURITY_ALLOW_DEFAULT_INIT_SECURITYINDEX, true).build();
+        ConfigurationRepository configRepository = createConfigurationRepository(settings);
+
+        var objectMapper = DefaultObjectMapper.objectMapper;
+        var objectNode = objectMapper.createObjectNode();
+        objectNode.set("_meta", objectMapper.createObjectNode().put("type", CType.ROLES.toLCString()).put("config_version", 2));
+        objectNode.set("kibana_read_only", objectMapper.createObjectNode().put("reserved", true));
+        objectNode.set("some_hidden_role", objectMapper.createObjectNode().put("hidden", true));
+        objectNode.set("all_access", objectMapper.createObjectNode().put("static", true)); // it reserved as well
+        objectNode.set("security_rest_api_access", objectMapper.createObjectNode().put("reserved", true));
+
+        configCache.put(
+            CType.ROLES,
+            SecurityDynamicConfiguration.fromJson(objectMapper.writeValueAsString(objectNode), CType.ROLES, 2, 1, 1)
+        );
+
+        SecurityDynamicConfiguration<?> config = configRepository.getConfiguration(CType.ROLES);
+        SecurityDynamicConfiguration<?> deepCloneConfig = configRepository.getConfiguration(CType.ROLES);
+
+        assertThat(config, is(notNullValue()));
+        assertThat(config, instanceOf(SecurityDynamicConfiguration.class));
+        // assertThat(config.getCEntries().size(), is(greaterThan(0)));
```

Comment:
nit: can remove this commented line

Reply:
@DarshitChanpura  I let this comment with the purpose of showing where the test is failing according to this [comment](https://github.com/opensearch-project/security/pull/3509#discussion_r1383470751). I would appreciate any help here.

Reply:
The issue here is that the cache you've created isn't connected to the cache used internally be the ConfigurationResponsitory - there is no access to it.  Rather than attempt to push that functionality in, I'd recommend wrapping `configRepository` with a spy, and then substituting your own invocation for a function call that would populate the cache, maybe `getConfigurationsFromIndex(...)` would be a good choice.

Reply:
Thanks for the tip @peternied. I'll try.

Reply:
I decided to test the `deepClone` method directly, so I remove the test from the ConfigRepository, and now both worked.
is this a good approach?
should I add more tests to `SecurityDynamicConfiguration<T>` class?  I didn't find any tests for this class, so I would like to know if it's worth it.
Please let me know your thoughts. @peternied